src/app/Http/Resources/Admin/MeetingChangeResource.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+namespace App\Http\Resources\Admin;
+
+use App\Http\Resources\JsonResource;
+use App\Repositories\FormatRepository;
+use App\Repositories\ServiceBodyRepository;
+use Illuminate\Support\Collection;
+
+class MeetingChangeResource extends JsonResource
+{
+    private static bool $isRequestInitialized = false;
+    private static Collection $allFormats;
+    private static Collection $allServiceBodies;
+
+    private static array $objectClassToStrMap = [
+        'c_comdef_meeting' => 'meeting',
+        'c_comdef_format' => 'format',
+        'c_comdef_user' => 'user',
+        'c_comdef_service_body'=> 'service_body',
+    ];
+
+    private static array $changeTypeToStrMap = [
+        'comdef_change_type_new' => 'created',
+        'comdef_change_type_delete' => 'deleted',
+        'comdef_change_type_change' => 'changed',
+        'comdef_change_type_rollback' => 'rolled_back',
+    ];
+
+    // Allows tests to reset state
+    public static function resetStaticVariables()
+    {
+        self::$isRequestInitialized = false;
+    }
+
+    public function toArray($request): array
+    {
+        if (!self::$isRequestInitialized) {
+            $this->initializeRequest($request);
+            self::$isRequestInitialized = true;
+        }
+
+        return [
+            'date_string' => date('g:i A, n/j/Y', strtotime($this->change_date)),
+            'user_name' => $this->user?->name_string ?? '',
+            'service_body_name' => $this->serviceBody?->name_string ?? '',
+            'details' => $this->getChangeDetailsString(),
+        ];
+    }
+
+    private function initializeRequest(): void
+    {
+        $formatRepository = new FormatRepository();
+        self::$allFormats = $formatRepository->search(showAll: true)->groupBy(['shared_id_bigint', 'lang_enum'], preserveKeys: true);
+
+        $serviceBodyRepository = new ServiceBodyRepository();
+        self::$allServiceBodies = $serviceBodyRepository->search()->mapWithKeys(fn ($sb) => [$sb->id_bigint => $sb]);
+    }
+
+    public function getChangeDetailsString(): string
```

Comment:
If this code is an exact copy of what is in the other resource, it should probably be moved into a base class that they both inherit.

Reply:
added ChangeDetailsTrait they both use